Source/WebCore/platform/ContentType.h

```diff
@@ -28,12 +28,18 @@
 
 #include <wtf/text/WTFString.h>
 
+namespace WTF {
+class URL;
+}
+
 namespace WebCore {
 
 class ContentType {
 public:
     WEBCORE_EXPORT explicit ContentType(String&& type);
     WEBCORE_EXPORT explicit ContentType(const String& type);
+    WEBCORE_EXPORT ContentType(const String& type, const WTF::URL&);
```

Comment:
Since the URL isn't stored, I think this makes more sense as a static method; something like:
```
    WEBCORE_EXPORT static fromURL(const WTF::URL&);
```

And the call site would change from:

```
        // Attempt to determine the content-type from the URL file extension.
        contentType = ContentType { emptyString(), url };
```

To:

```
        // Attempt to determine the content-type from the URL file extension.
        contentType = ContentType::fromURL(url);
```

Reply:
good idea.